scripts/lintcommit.lua

```diff
@@ -0,0 +1,131 @@
+function commit_message_is_ok(commit_message)
+  local commit_split = vim.split(commit_message, ":")
+
+  -- Return true if the type is vim-patch since most of the normal rules don't
+  -- apply.
+  if commit_split[1] == "vim-patch" then
+    return true
+  end
+
+  -- Return false if no colons are detected.
+  if vim.tbl_count(commit_split) < 2 then
+    return false
+  end
+
+  local before_colon = commit_split[1]
+  local after_colon = commit_split[2]
+
+  -- Check if commit introduces a breaking change.
+  if vim.endswith(before_colon, "!") then
+    before_colon = before_colon:sub(1, -2)
+  end
+
+  -- Check if type is correct
+  local type = vim.split(before_colon, "%(")[1]
+  local allowed_types = {"build", "ci", "docs", "feat", "fix", "perf", "refactor", "revert", "test", "chore"}
+  if not vim.tbl_contains(allowed_types, type) then
+    return false
+  end
+
+  -- Check if scope is empty
+  -- if before_colon:match("%(") then
+    -- scope = before_colon:match("%((.*)%)")
+  -- end
+
+  -- Check that description doesn't end with a period
+  if vim.endswith(after_colon, ".") then
+    return false
+  end
+
+  -- Check that description has exactly one whitespace after colon, followed by
+  -- a lowercase letter and then any number of letters.
+  if not string.match(after_colon, '^ %l%a*') then
+    return false
+  end
+
+  return true
+end
+
+function main()
+  local current_branch = vim.trim(vim.fn.system({'git', 'branch', '--show-current'}))
+  local ancestor_commit = vim.trim(vim.fn.system({'git', 'merge-base', 'origin/master', current_branch}))
+  local commits_str = vim.trim(vim.fn.system({'git', 'rev-list', ancestor_commit .. ".." .. current_branch}))
```

Comment:
Not sure this is a common case outside of maybe CI usage.

The last command `git rev-list ..` doesn't really produce anything on a shallow clone, but you can just re-use the `ancestor_commit` as a fallback. 

```console
$ git clone https://github.com/neovim/neovim neovim-test -- --depth=1
Cloning into 'neovim-test'...
remote: Enumerating objects: 2964, done.
remote: Counting objects: 100% (2964/2964), done.
remote: Compressing objects: 100% (2749/2749), done.
remote: Total 2964 (delta 208), reused 905 (delta 153), pack-reused 0
Receiving objects: 100% (2964/2964), 10.02 MiB | 6.78 MiB/s, done.
Resolving deltas: 100% (208/208), done.
$ cd neovim-test
$ git branch --show-current
master
$ git merge-base origin/master master
1f49268c46fcbe65f7e2e2cb620e6f51c059cf9e
$ git rev-list 1f49268c46fcbe65f7e2e2cb620e6f51c059cf9e..master # notice the empty output
$ git show -s --format=%s 1f49268c46fcbe65f7e2e2cb620e6f51c059cf9e # using the ancestor SHA again
refactor: convert TRUE/FALSE to true/false (#15660)
```

Reply:
Hmm, this shouldn't be a concern locally though, so I think it won't be needed for this PR. But I'll also keep this in mind, thank you!